### ISSUE TITLE
RecursionError in Restore Functionality for Related Fields in Recent Release

### ISSUE THREAD

**Issue by vardgeskarapetyan2** [san4ezy/django_softdelete#23]
Hello  all,

I've noticed a new release with updated logic regarding related fields. However, I've identified an issue here. I have two models related with a one-to-one field, as illustrated below.
`from django.db import models

class FirstModel(SoftDeletel):
    # Your fields for the first model go here

class SecondModel(SoftDeletel):
    first_model = models.OneToOneField(FirstModel, on_delete=models.CASCADE)
    # Other fields for the second model go here

`

When attempting to delete an instance of the Second Model, the deletion process proceeds without any issues. However, when attempting to restore, an error is encountered:

**RecursionError: maximum recursion depth exceeded while calling a Python object**

I debuged the flow and noticed that there is an issue with restore functioanlity.
Upon debugging the flow, I identified an issue within the restore functionality. When the restore() function is called for an instance of the Second Model, it restores it and invokes the restore function for the first_model field, given its one-to-one relationship. However, the problem arises when the restore function is called for the object of first_model. It, in turn, triggers the restoration of its related fields, including an instance of the Second Model, which has already been restored. Consequently, this leads to a maximum recursion count.

I appreciate your support in addressing this matter and hope for a prompt resolution. Thank you for your assistance, and I look forward to your response at your earliest convenience.
@san4ezy

**Comment by vardgeskarapetyan2** [comment]
The fix is smaller than I expected, I open a pull request(https://github.com/san4ezy/django_softdelete/pull/24/files).
Before restoring the associated field, delete_at must be set to the current value.
This will help to prevent calling restore for the same field multiple times:
  ```
  def __restore_related_object(self, field, related_object, strict, *args, **kwargs):
        """
        Restores a related object and saves it if it has a valid primary key.

        :param field: The related field.
        :type field: Any
        :param related_object: The related object to restore.
        :type related_object: Any
        :param strict: Flag to determine if restore should be strict or not.
        :type strict: bool
        :param args: Additional positional arguments for the related object's restore method.
        :type args: Any
        :param kwargs: Additional keyword arguments for the related object's restore method.
        :type kwargs: Any
        :return: None
        :rtype: None
        """
        if related_object.is_deleted:
            related_object.restore(strict=strict, *args, **kwargs)
            try:
                if related_object.pk is not None:
                    related_object.save()
            except AttributeError:
                pass
```

@san4ezy pls review the pull request, thank you in advance.

**Comment by san4ezy** [comment]
Hi @vardgeskarapetyan2 
Thank you for pointing out the problem and for the hot fix of it!
The PR is already approved and merged.
New version with your fix is 1.0.10

**Comment by vardgeskarapetyan2** [comment]
@san4ezy Thanks for your support and time. I'm closing the issue.
